src/components/configurable-columns/definition.js

```diff
@@ -0,0 +1,74 @@
+import { ConfigurableColumns } from './';
+import Definition from './../../../demo/utils/definition';
+import configurableColumnRowDefinition from './configurable-column-row/definition';
+
+let definition = new Definition('configurable-columns', ConfigurableColumns, {
+  associatedDefinition: [configurableColumnRowDefinition],
+  dataVariable: 'configurableColumnsData',
+  description: `Allow table columns to be configured.`,
+  designerNotes: `
```

Comment:
@benwilsonsage we'll need some UX input on the demo site text defined in this file